crates/ruff_linter/src/rules/flake8_errmsg/rules/string_in_exception.rs

```diff
@@ -281,23 +282,13 @@ fn generate_fix(
     exc_arg: &Expr,
     indentation: &str,
     stylist: &Stylist,
-    generator: Generator,
+    locator: &Locator,
 ) -> Fix {
-    let assignment = Stmt::Assign(ast::StmtAssign {
-        targets: vec![Expr::Name(ast::ExprName {
-            id: "msg".into(),
-            ctx: ExprContext::Store,
-            range: TextRange::default(),
-        })],
-        value: Box::new(exc_arg.clone()),
-        range: TextRange::default(),
-    });
-
     Fix::suggested_edits(
         Edit::insertion(
             format!(
-                "{}{}{}",
-                generator.stmt(&assignment),
+                "msg = {}{}{}",
+                locator.slice(exc_arg.range()),
```

Comment:
I assume this will cause syntax errors for multiline statements and other expressions that need to be parenthesized. Can you give that a try? E.g.:

```
raise ValueError(
    "foo"
    "bar"
)
```

Reply:
Ah yes, correct.

Reply:
We don't have a good general solution for this unfortunately. I'd love to have a method that takes a slice and returns `true` if it requires parentheses.

Reply:
Let me look at this tomorrow morning, can't think much now 😅